lyricsheets/models/karaoke.py

```diff
@@ -221,13 +429,25 @@ def _charKaraTimes(self) -> list[timedelta]:
             )
         ]
 
-        return list(accumulate(syllableCharLengths, operator.add, initial=self.start))
+        return list(accumulate(syllableCharLengths, operator.add, initial=self._start))
+
+    @property
+    def kLine(self) -> KLine:
+        return self.line
+    
+    @property
+    def kSyl(self) -> KSyl:
+        return self
+    
+    @property
+    def kChar(self) -> KChar:
+        return self.chars[0]
 
 
 @dataclass
-class KLine:
-    start: timedelta
-    end: timedelta
+class KLine(KObject):
+    _start: timedelta
+    _end: timedelta
```

Comment:
I get the idea of this, but it seems like everything outside the constructor for K* objects and the actual implementing method can just use `start` and `end` instead of the "internal" underscored versions?

(if I were to nitpick even more I'm also a little annoyed that the constructor now requires the use of the underscored variable name, maybe I'd even reimplement the constructor instead of letting the dataclass do it lol)

Reply:
There's a different implementation for `KChar`'s `start` and `end`, so I don't think that works out. I think I tried looking up if there was a way to have a property and data field of the same name that are both inherited from a parent class, but I couldn't find a way to do it smoothly, even if `KLine` and `KSyl` don't need it to be a property.

Reply:
True, every external access shouldn't use the underscored versions. Will change that.